mshadow/tensor_cpu-inl.h

```diff
@@ -309,7 +309,7 @@ inline void SoftmaxGrad(Tensor<cpu, 2, DType> dst,
 #pragma omp parallel for
   for (openmp_index_t y = 0; y < dst.size(0); ++y) {
     const int k = static_cast<int>(label[y]);
-    for (int x = 0; x < dst.size(1); ++x) {
+    for (int x = 0; x < static_cast<int>(dst.size(1)); ++x) {
```

Comment:
There was a problem here. This was my original suggestion, https://github.com/dmlc/mshadow/pull/275/commits/c39c0dc564b7642a1b92359198215057e2db84f5
I changed them to int based on comments in the PR. But that didn't solve all issues. I still think this above change is the right one. Labels can't be negative as per previous code right, because unsigned int k was used to store the label. Should I move the file back to the changes in above link?